storage/postgres/admin_storage.go

```diff
@@ -188,11 +201,71 @@ func (t *adminTX) GetTree(ctx context.Context, treeID int64) (*trillian.Tree, er
 }
 
 func (t *adminTX) ListTrees(ctx context.Context, includeDeleted bool) ([]*trillian.Tree, error) {
-	return nil, errNotImplemented
+	var query string
+	if includeDeleted {
+		query = selectTrees
+	} else {
+		query = selectNonDeletedTrees
+	}
+
+	stmt, err := t.tx.PrepareContext(ctx, query)
+	if err != nil {
+		return nil, err
+	}
+	defer stmt.Close()
+
+	rows, err := stmt.QueryContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	trees := []*trillian.Tree{}
+	for rows.Next() {
+		if err := rows.Err(); err != nil {
```

Comment:
You need to check `rows.Err()` after the loop (see the documentation for `rows.Next()`), since `rows.Next()` may return false either because there are no more rows, or an error occurred.